paasta_tools/paasta_remote_run.py

```diff
@@ -241,83 +334,210 @@ def task_config_to_dict(task_config):
+):
+    """ Executor stack consists of:
+    1. Cluster Executor (e.g. MesosExecutor)
+    2. LoggingExecutor
+    3. StatefulExecutor
+    """
+    # logging executor
+    task_logging_executor = processor.executor_from_config(
+        provider='logging',
+        provider_config={
+            'downstream_executor': cluster_executor,
+        },
+    )
+    # stateful executor
     StatefulExecutor = processor.executor_cls(provider='stateful')
     stateful_executor = StatefulExecutor(
         downstream_executor=task_logging_executor,
         persister=DynamoDBPersister(
-            table_name="taskproc_events_%s" % cluster,
-            session=session,
-            endpoint_url=endpoint,
+            table_name=f"taskproc_events_{cluster}",
+            session=create_boto_session(taskproc_config, region),
+            endpoint_url=taskproc_config.get('dynamodb_endpoint'),
         ),
     )
-
     return stateful_executor
 
 
+def set_runner_signal_handlers(runner):
+    def handle_interrupt(_signum, _frame):
+        paasta_print(
+            PaastaColors.red("Signal received, shutting down scheduler."),
+        )
+        if runner is not None:
+            runner.stop()
+        sys.exit(143 if _signum == signal.SIGTERM else 1)
+    signal.signal(signal.SIGINT, handle_interrupt)
+    signal.signal(signal.SIGTERM, handle_interrupt)
+
+
+def run_task(executor, task_config):
+    """ Runs a task until a terminal event is received, which is returned. """
+    runner = Sync(executor)
+    set_runner_signal_handlers(runner)
+    terminal_event = runner.run(task_config)
+    if getattr(terminal_event, 'platform_type', None) == 'lost':
+        runner.kill(task_config.task_id)
+    runner.stop()
+    return terminal_event
+
+
+def get_terminal_event_error_message(terminal_event):
+    if not terminal_event.terminal or terminal_event.success:
+        return None  # not terminal or not an error
+    elif TASKPROC_OFFER_TIMEOUT_RAW in terminal_event.raw:
+        return terminal_event.raw
+    else:
+        mesos_type = getattr(terminal_event, 'platform_type', None)
+        if mesos_type == 'failed':
+            error_message = ''
+        elif mesos_type == 'lost':
+            error_message = (
+                "- Task was lost probably due to a network partition or an "
+                "going away. It probably isn't coming back :("
+            )
+        elif mesos_type == 'error':
+            error_message = "- Encountered an unexpected error with Mesos"
+        else:
+            error_message = "- Unknown failure"
+
+        error_parts = [error_message] if error_message else []
+        error_parts.append(f"- Raw: {pprint.pformat(terminal_event.raw)}")
+        return '\n'.join(error_parts)
+
+
+def run_tasks_with_retries(executor_factory, task_config_factory, retries=0):
+    # use max in case retries is negative, +1 for initial try
+    tries_left = max(retries, 0) + 1
+    terminals = []
+
+    while tries_left > 0:
```

Comment:
can we use retrying executor from taskproc? exmaple: https://github.com/Yelp/task_processing/blob/master/examples/retry.py

we'd only have to wrap our executor stack and pass retries with task config dict.

Reply:
No problem.